### DISCUSSION TITLE
Dark mode themed colors

### DISCUSSION THREAD

**Discussion by napalmj** [tailwindlabs/tailwindcss#9108] [Ideas]
Currently, it seems dark mode is decided with the `dark:` prefix.
However, it would be nice if there is an option to configure a color theme for dark and light mode inside the `tailwind.config.js` file.

i.e.
```js
theme: {
  color-mode: {
    dark: {
      primary: black;
      secondary: purple;
    },
     light: {
       primary: white;
       secondary: gray;
    }
  }
}
```

or by just using the ternary operator

```js
theme: {
  colors: {
    primary: dark ? black : white;
    secondary: dark ? purple : gray;
  }
}
```

Is this something of which it will be okay to create a Pull Request for and is there any input for how we could make this better? This is just a rough draft idea to get the conversation started. 

Overall, it seems like a bad pattern to double write styles for `class="bg-white dark:bg-black`. Having something like `class="bg-primary"` is much cleaner, more scalable and requires less time when reworking styling. By implementing a themed styling for dark and light mode this will improve developer experience and increase the likelihood individuals will want to use tailwind.

**Comment by zslabs** [comment]
The dark: modifier is one way of doing this, but to cut down on additional classes, create a color palette that uses CSS variables that are reset to their dark variants based on a root .dark class. You'd then add the actual color definitions to the same file you add in the tailwind directives.

https://github.com/zslabs/zslabs-next/blob/master/tailwind.config.js

https://github.com/zslabs/zslabs-next/blob/master/styles/index.css
